gdraw/gmenu.c

```diff
@@ -1853,65 +1853,17 @@ static int osx_handle_keysyms( int st, int k )
 int osx_fontview_copy_cut_counter = 0;
 
 
-
-
-int GMenuBarCheckKey(GWindow top, GGadget *g, GEvent *event) {
-    int i;
-    GMenuBar *mb = (GMenuBar *) g;
-    GMenuItem *mi;
-    unichar_t keysym = event->u.chr.keysym;
-
+static int GMenuBarCheckHotkey(GWindow top, GGadget *g, GEvent *event) {
 //    TRACE("GMenuBarCheckKey(top) keysym:%d upper:%d lower:%d\n",keysym,toupper(keysym),tolower(keysym));
-
-    int SkipUnQualifiedHotkeyProcessing = 0;
-    // see if we should skip processing
-    if( g )
-    {
+    // see if we should skip processing (e.g. no modifier key pressed)
+    GMenuBar *mb = (GMenuBar *) g;
 	GWindow w = GGadgetGetWindow(g);
 	GGadget* focus = GWindowGetFocusGadgetOfWindow(w);
-	if( GGadgetGetSkipHotkeyProcessing(focus))
+	if (GGadgetGetSkipHotkeyProcessing(focus) || event->u.chr.state == 0)
 	    return 0;
-	SkipUnQualifiedHotkeyProcessing = GGadgetGetSkipUnQualifiedHotkeyProcessing(focus);
-    }
-
-
-    if ( g==NULL || keysym==0 ) return( false ); /* exit if no gadget or key */
-
-    if ( (menumask&ksm_cmdmacosx) && keysym>0x7f &&
-	    (event->u.chr.state&ksm_meta) &&
-	    !(event->u.chr.state&menumask&(ksm_control|ksm_cmdmacosx)) )
-	keysym = GGadgetUndoMacEnglishOptionCombinations(event);
 
-    if ( keysym<GK_Special && islower(keysym))
-	keysym = toupper(keysym);
-    if ( event->u.chr.state&ksm_meta && !(event->u.chr.state&(menumask&~(ksm_meta|ksm_shift)))) {
-	/* Only look for mneumonics in the leaf of the displayed menu structure */
-	if ( mb->child!=NULL )
-	    return( gmenu_key(mb->child,event)); /* this routine will do shortcuts too */
-
-	for ( i=0; i<mb->mtot; ++i ) {
-	    if ( mb->mi[i].ti.mnemonic == keysym && !mb->mi[i].ti.disabled ) {
-		GMenuBarKeyInvoke(mb,i);
-		return( true );
-	    }
-	}
-    }
 //    TRACE("GMenuBarCheckKey(2) keysym:%d upper:%d lower:%d\n",keysym,toupper(keysym),tolower(keysym));
 
-    /* First check for an open menu underscore key being pressed */
-    mi = GMenuSearchShortcut(mb->g.base,mb->mi,event,mb->child==NULL);
```

Comment:
This is the redundant block that was removed. Mnemonics are checked already here: https://github.com/fontforge/fontforge/pull/2844/files#diff-36bbf0dde11c76b3e8ff12e45265a75fR1995